main.py

```diff
@@ -0,0 +1,52 @@
+# flake8: noqa
+
+import argparse
+import json
+
+from . import spec, trie
+from .eth_types import Address, Uint
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Run a state transition.")
+
+    parser.add_argument("--input.alloc", dest="alloc", action="store")
+    parser.add_argument("--input.env", dest="env", action="store")
+    parser.add_argument("--input.txs", dest="txs", action="store")
+
+    args = parser.parse_args()
+
+    with open(args.alloc) as f:
+        alloc = json.load(f)
+    with open(args.env) as f:
+        env = json.load(f)
+
+    state = {}
+    for (addr, vals) in alloc.items():
+        account = spec.Account(
+            nonce=vals.get("nonce", Uint(0)),
+            balance=Uint(int(vals.get("balance", Uint(0)))),
+            code=bytes.fromhex(vals.get("code", "")),
+            storage={},  # TODO: support storage
+        )
+        addr = bytes.fromhex(addr)
+        state[addr] = account
+
+    gas_used, receipts, state = spec.apply_body(
+        state,
+        Address(bytes.fromhex(env["currentCoinbase"][2:])),
+        Uint(int(env["currentNumber"], 16)),
+        Uint(int(env["currentGasLimit"], 16)),
+        Uint(int(env["currentTimestamp"], 16)),
+        Uint(int(env["currentDifficulty"], 16)),
+        [],
+        [],
+    )
+
+    print(gas_used)
```

Comment:
We have had good success with using `logging` over `print` which has the nice trade off of a little bit more boilerplate code which yields the ability to have different granularity about what is displayed when the *thing* is run.

https://github.com/ethereum/ddht/blob/142911d134ff839f3f79ff8fe9e45d3fe5a58cd0/ddht/logging.py#L59-L77